hudi-common/src/main/java/org/apache/hudi/common/table/timeline/TimelineUtils.java

```diff
@@ -266,13 +266,19 @@ public static HoodieDefaultTimeline getTimeline(HoodieTableMetaClient metaClient
    * @return Hudi timeline.
    */
   public static HoodieTimeline getCommitsTimelineAfter(
-      HoodieTableMetaClient metaClient, String exclusiveStartInstantTime, Option<String> lastMaxCompletionTime) {
+      HoodieTableMetaClient metaClient,
+      String exclusiveStartInstantTime,
+      Option<String> lastMaxCompletionTime,
+      boolean shouldIncludeCleanerTimeline) {
```

Comment:
Typically we do not consider the cleaner as part of the commits since it is not changing the data. Is the issue here that the dropped partition from the replace commit is not being picked up when looking at the partitions affected for the meta sync?

Reply:
We do consider dropped partitions due to replacecommit in `TimelineUtils.getDroppedPartitions`. What was happening is that replacecommit does not add a new filegroup to a partition when all records are deleted. Older filegroups are not active and then subsequently cleaner thinks there are no active filegroups in the partition and marks the partition to be deleted. https://github.com/apache/hudi/blob/58bc859b173a3648ff5f7f2042aaadf8281cac2c/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanner.java#L353-L359

Reply:
Does it make sense to mark those partitions as deleted when the replace commit happens though or only when the cleaner kicks in? The benefit of doing it when the replace commit happens is that it will kick in before the files are actually removed. I'm guessing that the lack of files was causing some issue in a query engine. If that is the case then you will have some downtime with the current approach of doing this after the cleaner has removed the files.

Reply:
its not the lack of files. but cleaner deleted the physical partition dir, but we missed to sync to metastore and hence. but I see your point too. 
your suggestion is to reverse the ordering. if it goes into replace commit, first we will sync to metastore to drop the partition and later lazily cleaner will take care of deleting the dir physically. 

But w/ current approach, just after cleaner runs(after this fix), and before we run meta sync, if someone is querying, we might run into issues.

Reply:
Yes that is my concern.